var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -31,23 +31,41 @@ class Zfp(Package):
        target error bounds or bit rates.  Although bit-for-bit lossless
        compression is not always possible, zfp is usually accurate to
        within machine epsilon in near-lossless mode, and is often orders
-       of magnitude more accurate than other lossy compressors.
-
+       of magnitude more accurate than other lossy compressors. Versions
+       of zfp 0.5.1 or newer also support compression of integer data.
     """
 
     homepage = "http://computation.llnl.gov/projects/floating-point-compression"
-    url      = "http://computation.llnl.gov/projects/floating-point-compression/download/zfp-0.5.0.tar.gz"
+    url      = "http://computation.llnl.gov/projects/floating-point-compression/download/zfp-0.5.1.tar.gz"
 
+    version('0.5.1', '0ed7059a9b480635e0dd33745e213d17')
     version('0.5.0', '2ab29a852e65ad85aae38925c5003654')
 
+    variant('bswtuint8', default=False,
+            description='Build with bit stream word type of uint8')
+
+    def edit(self, spec, prefix):
```

Comment:
The `edit` phase isn't actually run unless you convert this to the `MakefilePackage` base class. Then you would also want to split the `install` phase into `build` and `install`.